libs/server-sent-events/src/client.cpp

```diff
@@ -276,6 +320,7 @@ class FoxyClient : public Client,
     }
 
     void do_shutdown(std::function<void()> completion) {
```

Comment:
This was causing unnecessary hanging when the test service tells the client to close.